cpp/oneapi/dal/algo/logistic_regression/backend/gpu/infer_kernel_dense_batch_dpc.cpp

```diff
@@ -0,0 +1,143 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/detail/profiler.hpp"
+
+#include "oneapi/dal/backend/interop/common.hpp"
+#include "oneapi/dal/backend/interop/common_dpc.hpp"
+#include "oneapi/dal/backend/interop/error_converter.hpp"
+#include "oneapi/dal/backend/interop/table_conversion.hpp"
+
+#include "oneapi/dal/backend/dispatcher.hpp"
+#include "oneapi/dal/backend/primitives/blas.hpp"
+#include "oneapi/dal/backend/primitives/ndarray.hpp"
+#include "oneapi/dal/backend/primitives/objective_function.hpp"
+#include "oneapi/dal/backend/primitives/ndindexer.hpp"
+
+#include "oneapi/dal/table/row_accessor.hpp"
+
+#include "oneapi/dal/algo/logistic_regression/common.hpp"
+#include "oneapi/dal/algo/logistic_regression/train_types.hpp"
+#include "oneapi/dal/algo/logistic_regression/backend/model_impl.hpp"
+#include "oneapi/dal/algo/logistic_regression/backend/gpu/infer_kernel.hpp"
+
+namespace oneapi::dal::logistic_regression::backend {
+
+using daal::services::Status;
+using dal::backend::context_gpu;
+
+namespace be = dal::backend;
+namespace pr = be::primitives;
+namespace interop = dal::backend::interop;
+
+template <typename Float>
+std::int64_t propose_block_size(const sycl::queue& q, std::int64_t f, std::int64_t r) {
+    constexpr std::int64_t fsize = sizeof(Float);
+    return 0x10000l * (8 / fsize);
+}
+
+template <typename Float, typename Task>
+static infer_result<Task> call_dal_kernel(const context_gpu& ctx,
```

Comment:
I suggest to avoid using `call_dal_kernel` name

Reply:
We have it in many algorithms (in linear_regression, for example) so I do not understand what is the problem with naming

Reply:
Because this name is not representative in terms of what this kernel does